src/shared/errors.ts

```diff
@@ -0,0 +1,78 @@
+const errorFactory = (templ: (...prop: string[]) => string) => new Proxy({}, {
+  get(_, prop: string) {
+    return templ(...prop.replace('_', ' ').split(','));
+  },
+  has(_, prop) {
+    return true;
+  }
+});
+
+const errors = {
```

Comment:
Maybe just bite the bullet and use an i18n library for this, it's not that much extra work, and we get interpolation in translated strings for free, pretty much.

Reply:
I've taken a look at the i18n libraries around, and to be honest the overall benefit seems marginal -- what we gain in convenience, we lose in package size. Can be something to experiment with in the future though, I think.